h5netcdf/core.py

```diff
@@ -123,6 +122,21 @@ def __getitem__(self, key):
     def __setitem__(self, key, value):
         self._h5ds[key] = value
 
+    def resize(self, size, axis=None):
```

Comment:
Thinking about this a little more, I'm concerned that this could allow for creating netCDF files with inconsistent dimension sizes between variables. This shouldn't be possible with h5netcdf's public API.

Can we move this to a `Group` level instead, and resize all variables contained inside the group that use the dimension at once? The signature would look something like `group.resize(size, dimension)` where `dimension` is the name of the dimension to be resized.

Reply:
Sure.